src/haddock/modules/analysis/ilrmsdmatrix/__init__.py

```diff
@@ -205,29 +214,58 @@ def _run(self):
             nmodels,
             tot_npairs,
             ncores)
-
-        # Calculate the ilrmsd for each set of models
-        ilrmsd_jobs = []
-        self.log(f"running ilRmsd Jobs with {ncores} cores")
+        
+        rec_traj_filename = "traj_rec.xyz"
+        lig_traj_filename = "traj_lig.xyz"
+        prev_keys = []
+        with open(rec_traj_filename, "w") as rec_traj_xyz:
+            with open(lig_traj_filename, "w") as lig_traj_xyz:
+                for mod in models:
+                    atoms = {}
+                    atoms.update(get_atoms(mod))
```

Comment:
Bit confusing here imo

```suggestion
                    atoms: AtomsDict = get_atoms(mod)
```